### PR TITLE
Making an admin page

### PR DESCRIPTION
**Pull request by jasminechau2**
Just the basic prototype for the admin page does not do anything right now besides displaying text. Do not worry will work on it more!

screen shot 2018-05-18 at 1 06 40 am

### CODE REVIEW

File: src/main/java/codeu/controller/AdminServlet.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package codeu.controller;
+
+import java.io.IOException;
```
Comment (yebrahim) [inline src/main/java/codeu/controller/AdminServlet.java]: Can we review this list for unneeded imports?

File: src/main/java/codeu/controller/AdminServlet.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import org.jsoup.safety.Whitelist;
+
+/** Servlet class responsible for the Admin page. */
+public class AdminServlet extends HttpServlet {
+
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+    throws IOException, ServletException {
+      request.getRequestDispatcher("/WEB-INF/view/admin.jsp").forward(request, response);
+
+}
```
Comment (yebrahim) [inline src/main/java/codeu/controller/AdminServlet.java]: nitpick: indentation.

File: src/main/webapp/WEB-INF/view/admin.jsp [hunk 3]
```diff
@@ -0,0 +1,26 @@
+
+<!DOCTYPE html>
+<html>
+	<head>
```
Comment (yebrahim) [inline src/main/webapp/WEB-INF/view/admin.jsp]: @jasminechau2 @vocatwk @lealgamb:
There's a mix of tabs and spaces for indentation in this file, which results in indentation looking incorrect.
Even though it's (mostly) personal preference, open source projects often pick a convention and stick to it. You can then change your editors settings to use one or the other and format the entire file accordingly. :)

**Comment by yebrahim** [comment]
Also, can you edit the PR's title so it's more descriptive?

**Comment by jasminechau2** [comment]
Some of these commits were made after I fixed a small error I am sorry for the large amount of commits

File: src/main/webapp/WEB-INF/view/register.jsp [hunk 4]
```diff
@@ -49,6 +49,8 @@
       <br/>
       <input type="password" name="password" id="password">
       <br/><br/>
+      <button name="button" value="true" type="button">Admin</button>
```
Comment (vocatwk) [inline src/main/webapp/WEB-INF/view/register.jsp]: i'm a bit confused on what this button is going to do?

File: src/main/java/codeu/controller/AdminServlet.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    request.getRequestDispatcher("/WEB-INF/view/admin.jsp").forward(request, response);
+
+    String username = request.getParameter("username");
+
+    User user = userStore.getUser(username);
+    if(user.getAdmin() == false){
```
Comment (vocatwk) [inline src/main/java/codeu/controller/AdminServlet.java]: I think there is a nullPointerException here

File: src/main/java/codeu/controller/AdminServlet.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    this.ConversationStore = ConversationStore;
+  }
+  /**
+   * This function fires when a user requests the /login URL. It simply forwards the request to
+   * login.jsp.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    request.getRequestDispatcher("/WEB-INF/view/admin.jsp").forward(request, response);
```
Comment (vocatwk) [inline src/main/java/codeu/controller/AdminServlet.java]: is this supposed to be here?

File: src/main/java/codeu/controller/AdminServlet.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+/** Servlet class responsible for the Admin page. */
+public class AdminServlet extends HttpServlet {
+
+  private UserStore userStore;
+  private ConversationStore ConversationStore;
+  private MessageStore MessageStore;
+
+  /**
+   * Set up state for handling login-related requests. This method is only called when running in a
```
Comment (yebrahim) [inline src/main/java/codeu/controller/AdminServlet.java]: Fix the comment so it doesn't talk about the login page.

File: src/main/java/codeu/controller/AdminServlet.java [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+
+  void setConversationStore(ConversationStore ConversationStore) {
```
Comment (yebrahim) [inline src/main/java/codeu/controller/AdminServlet.java]: Missing method comment, here and below.

File: src/main/java/codeu/controller/AdminServlet.java [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+      throws IOException, ServletException {
+     String username = request.getParameter("username");
+
+/* checks to see if user is admin if false redrects them to the login page*/
+    User user = userStore.getUser(username);
+    if(user.getAdmin() == false){
+      response.sendRedirect("/login");
+      return;
+    }else if (user.getAdmin() == true){
+      /* an attempt to grab information from the stores to display on the page */
```
Comment (yebrahim) [inline src/main/java/codeu/controller/AdminServlet.java]: Do we need this comment?

File: src/main/java/codeu/controller/AdminServlet.java [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+    int numOfUsers = userStore.getAllUsers().size();
```
Comment (yebrahim) [inline src/main/java/codeu/controller/AdminServlet.java]: Can we fix the indentation for these lines and the braces in the end?

File: src/main/java/codeu/controller/AdminServlet.java [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import org.mindrot.jbcrypt.BCrypt;
+
+/** Servlet class responsible for the Admin page. */
+public class AdminServlet extends HttpServlet {
+
+  private UserStore userStore;
+  private ConversationStore ConversationStore;
```
Comment (yebrahim) [inline src/main/java/codeu/controller/AdminServlet.java]: The instance name should have camelCase, and it should be different from the class name. It's interesting that the Java compiler didn't complain about this. :)

File: src/main/java/codeu/controller/AdminServlet.java [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+      response.sendRedirect("/login");
+      return;
+    }else if (user.getAdmin() == true){
+      /* an attempt to grab information from the stores to display on the page */
+
+    int numOfUsers = userStore.getAllUsers().size();
+
+    int numOfConvos = ConversationStore.getAllConversations().size();
+
+    int Messages = MessageStore.getMessages().size();
```
Comment (yebrahim) [inline src/main/java/codeu/controller/AdminServlet.java]: Can we call this `numOfMessages` like you did above? Or maybe `messagesLength`, or `numMessages`.

Comment (yebrahim) [reply src/main/java/codeu/controller/AdminServlet.java]: It'd also be nice if we can rename the string attributes below to add an indication that these are the "sizes" not the actual list.

File: src/main/java/codeu/controller/RegisterServlet.java [hunk 13]
```diff
@@ -67,6 +67,15 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     User user = new User(UUID.randomUUID(), username, hashedPassword, Instant.now());
     userStore.addUser(user);
 
+    String button = request.getParameter("adminbutton");
```
Comment (yebrahim) [inline src/main/java/codeu/controller/RegisterServlet.java]: Is this just for the sake of the prototype? We probably don't want all users to be able to sign up as admins. Maybe we should start with a hardcoded list of usernames?

File: src/main/java/codeu/model/store/basic/MessageStore.java [hunk 14]
```diff
@@ -103,4 +103,9 @@ public void addMessage(Message message) {
   public void setMessages(List<Message> messages) {
     this.messages = messages;
   }
+
+  public List<Message> getMessages() {
```
Comment (yebrahim) [inline src/main/java/codeu/model/store/basic/MessageStore.java]: Method documentation.

File: src/main/webapp/WEB-INF/view/admin.jsp [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+    <a href="/conversations">Conversations</a>
+    <% if(request.getSession().getAttribute("user") != null){ %>
+      <a href="/profile/<%=request.getSession().getAttribute("user") %>">
+         Hello <%= request.getSession().getAttribute("user") %>!</a>
+    <% } else{ %>
+      <a href="/login">Login</a>
+    <% } %>
+    <a href="/about.jsp">About</a>
+  </nav>
+	<form action="/login" method="POST">
```
Comment (yebrahim) [inline src/main/webapp/WEB-INF/view/admin.jsp]: I think we can start by getting the statistics in the page with the GET request directly if the username is in a given list.

**Comment by yebrahim** [comment]
You were going about it the right way. I'd start with reverting the last changes that add persistence, and move the logic in `doPost` to `doGet`. Afterwards, we can reduce the logic in `doGet` to just be computing the statistics and passing them to the JSP.

**Comment by yebrahim** [comment]
Should we close this PR in favor of #35?

**Comment by jasminechau2** [comment]
yes

**Pull request closed without merging**